drivers/timer/nrf_rtc_timer.c

```diff
@@ -88,6 +88,9 @@ static void prevent_false_prev_evt(void)
 		k_busy_wait(15);
 		event_clear();
 	}
+
+	/* If previous compare event already triggered an interrupt. */
+	NVIC_ClearPendingIRQ(RTC1_IRQn);
```

Comment:
Instead of `RTC1_IRQn` hard-coded here it would be good to use a macro defined next to `RTC` in [line 16](https://github.com/zephyrproject-rtos/zephyr/pull/25517/files#diff-57dfbf11ae1437d7b0a30d68d709d8d6L16). Theoretically, someone might want to change the RTC instance used by this driver.

Reply:
it was copy from init function. I've removed RTC1 dependency from the code (second commit).